### PR TITLE
Added mosquitto e cjson packages

### PR DESCRIPTION
**Pull request by danielecesarini**
This PR adds [mosquitto](https://mosquitto.org/) e [cjson](https://github.com/DaveGamble/cJSON) packages to SPACK

### CODE REVIEW

File: var/spack/repos/builtin/packages/mosquitto/package.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+        if '~tls' in self.spec:
+            args.append('-DWITH_TLS=no')
+
+        if '+static' in self.spec:
+            args.append('-DWITH_STATIC_LIBRARIES=yes')
+
+        if '+websocket' in self.spec:
+            args.append('-DWITH_WEBSOCKETS=yes')
+
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/mosquitto/package.py]:
```suggestion
        args = [
            self.define('DOCUMENTATION', 'no'),
            self.define_from_variant('WITH_CJSON', 'cjson'),
            self.define_from_variant('WITH_TLS', 'tls'),
            self.define_from_variant('WITH_STATIC_LIBRARIES', 'static'),
            self.define_from_variant('WITH_WEBSOCKETS', 'websocket')
        ]
```

**Review by alalazo — CHANGES_REQUESTED** [review]
Do we need all the versions listed here, or being new packages we could be fine only with the latest release?

**Comment by danielecesarini** [comment]
> Do we need all the versions listed here, or being new packages we could be fine only with the latest release?

I have some codes that need the support of old versions.

**Comment by alalazo** [comment]
Can we get by with the latest versions of the minor releases you need?

**Comment by danielecesarini** [comment]
> Can we get by with the latest versions of the minor releases you need?

[Fix](https://github.com/spack/spack/pull/28550/commits/5e47a56d8cfd2ab82e0954ac3bfe3d48a383ea84l)

**Comment by danielecesarini** [comment]
@alalazo what do you think?

**Comment by alalazo** [comment]
@danielecesarini There's a suggestion for more idiomatic syntax that has not been addressed at https://github.com/spack/spack/pull/28550#discussion_r790250090 Also, I think we can remove a lot of old patch releases. Do we need all the 15 releases in the 1.7 series or can we just use the latest (since it is a new package)?

**Comment by danielecesarini** [comment]
> @danielecesarini There's a suggestion for more idiomatic syntax that has not been addressed at [#28550 (comment)](https://github.com/spack/spack/pull/28550#discussion_r790250090) Also, I think we can remove a lot of old patch releases. Do we need all the 15 releases in the 1.7 series or can we just use the latest (since it is a new package)?

My apologies, I had not realized.

I removed older versions, but unfortunately I need at least the version 1.3.5 for a legacy code. My rationale is to keep old major versions down to 1.3.5, which is my dependency.

I also updated the source code repository targeting the official archive repository of mosquitto.

**Review by alalazo — DISMISSED** [review]

**Comment by danielecesarini** [comment]
@alalazo I am sorry, I forgot an important acceleration library for mosquitto. I've just added to this PR.

**Review by alalazo — APPROVED** [review]

**Pull request merged**